### PR TITLE
[#6791] allow both regex- and glob-style matching (4-3-stable)

### PR DESCRIPTION
**Pull request by d-w-moore**
Corresponding to https://github.com/irods/irods/pull/6946

### CODE REVIEW

File: src/itree.cpp [hunk 1]
```diff
@@ -13,16 +13,24 @@
 #include <boost/program_options.hpp>
 #include <nlohmann/json.hpp>
 
+#include <cassert>
```
[the remark this answers is not in the document]
Comment (d-w-moore) [reply src/itree.cpp]: Looks like it's not... deleting.

**Comment by alanking** [comment]
Is this ready for review?

**Comment by d-w-moore** [comment]
Yes, but it's basically just the equivalent commit from #369 in [main](https://github.com/d-w-moore/irods_client_icommands/commits/6791.m).  So, squash and # then ?

File: src/itree.cpp [hunk 2]
```diff
@@ -31,13 +38,15 @@ auto print_dir(const fs::path&,
 auto correct_path(const po::variables_map&, rodsEnv&) -> fs::path;
 auto print_usage() -> void;
 auto get_json(const fs::path&, const po::variables_map&, ix::client_connection&, unsigned int depth) -> json::json;
+auto contains_pattern(const pattern_matcher& pm) -> bool;
+
```
Comment (alanking) [inline src/itree.cpp:42]: I know, I know...

I can't help but point it out here, too :)

**Review by alanking — COMMENTED** [review]
Yep. Nix the extra line, squash and #.

File: src/itree.cpp (hunk 2 above)
Comment (d-w-moore) [reply src/itree.cpp:42]: (y)

**Comment by d-w-moore** [comment]
is now #'d.

**Review by alanking — APPROVED** [review]

**Pull request merged**